front/src/applications/operationalStudies/components/MacroEditor/OsrdNgeSync.tsx

```diff
@@ -683,92 +685,44 @@ export default class OsrdNgeSync {
    */
-  private toGraph(): {
-    nodes: (ColaNode & { key: string })[];
-    edges: ColaLink<number>[];
-  } {
-    const nodes = uniqBy(
-      this.trainSchedules.map((ts) => ts.path).flat(),
+  private applyLayout() {
+    const indexedNodes = uniqBy(
+      this.trainSchedules.flatMap((ts) => ts.path),
       OsrdNgeSync.getPathKey
     ).map((pathItem) => {
       const key = OsrdNgeSync.getPathKey(pathItem);
-      const nodeIndexed = this.getNodeByKey(key);
-      const fixed = nodeIndexed!.saved ? 1 : 0;
-      return {
-        key,
-        x: nodeIndexed!.node.position_x,
-        y: nodeIndexed!.node.position_y,
-        height: 100,
-        width: 50,
-        fixed,
-      };
+      return this.getNodeByKey(key)!;
     });
 
-    const nodeIndexByKey = nodes.reduce(
-      (acc, curr, index) => {
-        acc[curr.key] = index;
-        return acc;
-      },
-      {} as Record<string, number>
-    );
-
-    const edges: { source: number; target: number; weight: number }[] = [];
-    const edgesIndex: Record<string, number> = {};
-    this.trainSchedules.forEach((ts) => {
-      let prevInNodeArrayIndex: number | null = null;
-      ts.path.forEach((item) => {
-        // create edges
-        const targetKey = OsrdNgeSync.getPathKey(item);
-        const targetNodeIndexInNodeArray = nodeIndexByKey[targetKey];
-        if (prevInNodeArrayIndex) {
-          const edgeIndexKey = `${prevInNodeArrayIndex}-${targetNodeIndexInNodeArray}`;
-          if (edgesIndex[edgeIndexKey] !== undefined) {
-            edges[edgesIndex[edgeIndexKey]].weight += 1;
-          } else {
-            const newLength = edges.push({
-              source: prevInNodeArrayIndex,
-              target: targetNodeIndexInNodeArray,
-              weight: 1,
-            });
-            edgesIndex[edgeIndexKey] = newLength - 1;
-          }
-        }
-        prevInNodeArrayIndex = targetNodeIndexInNodeArray;
-      });
-    });
-
-    return {
-      nodes,
-      edges,
-    };
-  }
-
-  /**
-   * Apply a layout on nodes and save the new position.
-   * Nodes that are saved are fixed.
-   */
-  private applyLayout() {
-    const graph = this.toGraph();
-    if (graph.nodes.some((n) => !n.fixed)) {
-      const layout = new Layout()
-        .defaultNodeSize(100)
-        .size([800, 600])
-        .linkDistance(200)
-        .avoidOverlaps(true)
-        .nodes(graph.nodes)
-        .links(graph.edges)
-        .start(5, 10, 15, 20, false, false);
-
-      layout.nodes().forEach((n) => {
-        const node = n as ColaNode & { key: string };
-        this.updateNodeDataByKey(node.key, {
-          position_x: Math.round(node.x),
-          position_y: Math.round(node.y),
+    const geoNodes = indexedNodes.filter((n) => n.node.geocoord);
+    const xCoords = geoNodes.filter((n) => n.node.geocoord).map((n) => n.node.geocoord!.lng);
+    const yCoords = geoNodes.filter((n) => n.node.geocoord).map((n) => n.node.geocoord!.lat);
```

Comment:
You can remove the filter since you already do it l700
```suggestion
    const xCoords = geoNodes.map((n) => n.node.geocoord!.lng);
    const yCoords = geoNodes.map((n) => n.node.geocoord!.lat);
```